test/apps/iodemo/io_demo.cc

```diff
@@ -408,6 +425,8 @@ class DemoClient : public P2pDemoCommon {
             _io_counter      = io_counter;
             _conn_io_counter = conn_io_counter;
             _chunk_cnt       = chunk_cnt;
+            _counters.add_counter(_io_counter);
+            _counters.add_counter(_conn_io_counter);            
```

Comment:
malloc on fast path

Reply:
@yosefe Do you mean malloc ``_counters``?

Reply:
yes, the vector

Reply:
why fast path? it should be only on connect/reconnect, also we can reserve memory once since we know number of counters

Reply:
IoReadResponseCallback::init is called on fast path
we should just hold 2 pointers like in prev version of this PR

Reply:
but at this point the object has to be created and initialized, it can be placed in `server_info_t`, std::vector is just at example, we can hold 2 pointers (until no need more)